extensions/amp-ad/0.1/amp-ad.js

```diff
@@ -69,11 +69,28 @@ export class AmpAd extends AMP.BaseElement {
 
       this.win.ampAdSlotIdCounter = this.win.ampAdSlotIdCounter || 0;
       const slotId = this.win.ampAdSlotIdCounter++;
+      let slotIdByUnit;
+
+      if (this.element.hasAttribute('data-slot')) {
+        const dataSlot = this.element.getAttribute('data-slot');
+        this.win.ampAdSlotIdByUnitCounter =
+          this.win.ampAdSlotIdByUnitCounter || {};
+        this.win.ampAdSlotIdByUnitCounter[dataSlot] =
+          this.win.ampAdSlotIdByUnitCounter[dataSlot] || 0;
+        slotIdByUnit = this.win.ampAdSlotIdByUnitCounter[dataSlot]++;
+      }
 
       return new Promise((resolve) => {
         this.getVsync().mutate(() => {
           this.element.setAttribute('data-amp-slot-index', slotId);
 
+          if (this.element.hasAttribute('data-slot')) {
+            this.element.setAttribute(
+              'data-vars-slot-index-by-unit',
+              slotIdByUnit
```

Comment:
```suggestion
              'data-amp-unit-slot-index',
              unitSlotId
```

Reply:
@powerivq the goal to use the `data-vars` pattern is to be able to access this value in `amp-analytics` tag, could we move forward with `data-vars`?

Reply:
@powerivq I will check it before continue working in this PR